src/conf.c

```diff
@@ -354,6 +354,8 @@ int ConfGetChildValue(const ConfNode *base, const char *name, const char **vptr)
         return 0;
     }
     else {
+        if (node->val == NULL)
+            return 0;
```

Comment:
I don't think this is correct, even tho it might not break anything.  `NULL` is a valid value for a configuration parameter:

```
some-value: ~
```

In which case I think we'd want the output pointer to be set to `NULL` given the node was found.

Reply:
Could use a unit test for this.

Reply:
What would be the use case for allowing a NULL value?

Reply:
The way I interpret Jason's comment is:
Aim of the function: Find out whether the node exists (1) or not (0).
According to this PR's way, on returning 0, I'm claiming that the node didn't exist at all when in fact it did exist but just did not have any value assigned to it.

Reply:
"Value" returning a NULL pointer as a valid "value" seems a bit like an anti-pattern to me, but I'm open to changing my mind if there are use cases.

Reply:
@jasonish do you have an example in mind?

Reply:
Given the YAML:

```
parent:
  child: ~
```

`child` exists with a value of NULL. Given a handle on the parent node,

```
const char *value = "bad";
ConfGetChildValue(parent, "child", &value);
```

I'd expect the function to return truthy (1) as child does exist, and have a NULL value.

Reply:
> What is the meaning of `child: ~`?

YAML for `NULL`. Other NULL values are `null` and no value, such as `value:`.

Reply:
Do we have to do anything? See this updated unit test: 

https://github.com/OISF/suricata/compare/master...jasonish:suricata:conf-get-child-value-null/v1

Maybe there is an expectation here that this function should fail out if its not a string?  Which is fair, but should be reflected with some function documentation.

Reply:
w/o this PR, the af-packet parsing code segv's since it doesn't check the ptr

Reply:
this can of course be fixed by adding a ptr null check, but that is what started this discussion. Why check a ptr if the function returns success.

Reply:
Valid point. From a purely YAML perspective, NULL is a valid value on success. So maybe function documentation to make it clear its more like a function named `ConfGetChildValueAsString`... And a unit test. Without that, a field that exists and is intentionally NULL shouldn't return an error IMO.

Serde makes this very clear, where the code would be something like...
- get value, returns true if value exists
- now you have to test what type of value it is, NULL, a string, a number?  Due to the enum type returned you are forced to do this extra check.